R/quantity.R

```diff
@@ -127,7 +136,14 @@ Quantity <- R6::R6Class(
     #' @param unit Unit to check
     hasUnit = function(unit) {
       validateIsString(unit)
-      rClr::clrCallStatic(WITH_DIMENSION_EXTENSION, "HasUnit", self$ref, unit)
+      unit %in% self$allUnits

```

Comment:
Here we do not use the .NET method. Instead we just check in the local array that is allocated only once